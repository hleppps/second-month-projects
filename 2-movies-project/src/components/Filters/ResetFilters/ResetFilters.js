import { Button } from "react-bootstrap";

const ResetFilters = (props) => {
  const onClickHandler = () => {
    props.resetFilters();
  };

  return (
    <Button
      data-testid="reset-button"
      className="w-100"
      variant="outline-secondary"
      onClick={onClickHandler}
    >
      Reset Filters
    </Button>
  );
};

export default ResetFilters;
