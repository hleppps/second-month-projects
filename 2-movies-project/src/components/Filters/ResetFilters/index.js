import ResetFilters from "./ResetFilters";

export default ResetFilters;
