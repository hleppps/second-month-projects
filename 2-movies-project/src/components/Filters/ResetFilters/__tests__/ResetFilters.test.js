import { render, cleanup, fireEvent } from "@testing-library/react";
import ResetFilters from "../ResetFilters";

afterEach(cleanup);

describe("ResetFilters component", () => {
  it("calls the year change function on change", () => {
    const resetfiltersHandler = jest.fn();

    const { getByTestId } = render(
      <ResetFilters resetFilters={resetfiltersHandler} />
    );

    fireEvent.click(getByTestId("reset-button"));

    expect(resetfiltersHandler).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<ResetFilters />);
    expect(asFragment(<ResetFilters />)).toMatchSnapshot();
  });
});
