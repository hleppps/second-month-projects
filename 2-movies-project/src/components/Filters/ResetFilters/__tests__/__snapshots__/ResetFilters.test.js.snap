// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetFilters component matches a snapshot 1`] = `
<DocumentFragment>
  <button
    class="w-100 btn btn-outline-secondary"
    data-testid="reset-button"
    type="button"
  >
    Reset Filters
  </button>
</DocumentFragment>
`;
